src/nnue/evaluate_nnue.cpp

```diff
@@ -122,39 +133,57 @@ static bool write_header(std::ostream& stream, std::uint32_t hashValue, const st
 }
 
 // Read network parameters
-static bool read_parameters(std::istream& stream) {
+static bool read_parameters(std::istream& stream, bool small) {
 
     std::uint32_t hashValue;
-    if (!read_header(stream, &hashValue, &netDescription))
+    if (!read_header(stream, &hashValue, &netDescription[small]))
+        return false;
+    if (hashValue != HashValue[small])
         return false;
-    if (hashValue != HashValue)
+    if (!small && !Detail::read_parameters(stream, *featureTransformerBig))
         return false;
-    if (!Detail::read_parameters(stream, *featureTransformer))
+    if (small && !Detail::read_parameters(stream, *featureTransformerSmall))
         return false;
     for (std::size_t i = 0; i < LayerStacks; ++i)
-        if (!Detail::read_parameters(stream, *(network[i])))
+    {
+        if (!small && !Detail::read_parameters(stream, *(networkBig[i])))
             return false;
+        if (small && !Detail::read_parameters(stream, *(networkSmall[i])))
+            return false;
+    }
     return stream && stream.peek() == std::ios::traits_type::eof();
 }
 
 // Write network parameters
-static bool write_parameters(std::ostream& stream) {
+static bool write_parameters(std::ostream& stream, bool small) {
 
-    if (!write_header(stream, HashValue, netDescription))
+    if (!write_header(stream, HashValue[small], netDescription[small]))
         return false;
-    if (!Detail::write_parameters(stream, *featureTransformer))
+    if (!small && !Detail::write_parameters(stream, *featureTransformerBig))
+        return false;
+    if (small && !Detail::write_parameters(stream, *featureTransformerSmall))
         return false;
     for (std::size_t i = 0; i < LayerStacks; ++i)
-        if (!Detail::write_parameters(stream, *(network[i])))
+    {
+        if (!small && !Detail::write_parameters(stream, *(networkBig[i])))
+            return false;
+        if (small && !Detail::write_parameters(stream, *(networkSmall[i])))
             return false;
+    }
     return bool(stream);
 }
 
 void hint_common_parent_position(const Position& pos) {
-    featureTransformer->hint_common_access(pos);
+
+    int simpleEval = pos.simple_eval();
+    if (abs(simpleEval) > 1100)
+        featureTransformerSmall->hint_common_access(pos);
+    else
+        featureTransformerBig->hint_common_access(pos);
 }
 
 // Evaluation function. Perform differential calculation.
+template<bool Small>
```

Comment:
As said earlier, it should be an enum. Boolean flags are undecypherable from the callsite.

Reply:
Done